basiclti/tsugi-util/src/java/org/tsugi/time/InstantUtil.java

```diff
@@ -0,0 +1,65 @@
+package org.tsugi.time;
+
+import java.util.Date;
+import java.time.Instant;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeParseException;
+
+import org.apache.commons.httpclient.util.DateUtil;
+import org.apache.commons.httpclient.util.DateParseException;
+
+public class InstantUtil {
+
+	public static Instant parseGMTFormats(String dateString)
+	{
+		Instant retval = null;
+		Date d = null;
+		SimpleDateFormat format = null;
+	
+		// https://docs.oracle.com/javase/10/docs/api/java/time/Instant.html#parse(java.lang.CharSequence)
+		// "2007-12-03T10:15:30.00Z"
+		try {
+			retval = Instant.parse(dateString);
+			if ( retval != null ) return retval;
+		} catch (DateTimeParseException e) {
+			// Ignore
+		}
+
+		// https://stackoverflow.com/questions/1930158/how-to-parse-date-from-http-last-modified-header
+		// "Wed, 09 Apr 2008 23:55:38 GMT"
+		try {
+			format = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss zzz");
+			d = format.parse(dateString);
+			if ( d != null && d.toInstant() != null ) return d.toInstant();
+		} catch (Exception e) {
+			// Ignore
```

Comment:
instead of ignoring maybe a debug statement?

Reply:
So - this is in 'tsugi-util' that is trying to keep its external dependencies to an absolute minimum.  Historically I have not linked the code in this folder to *anything* that is Sakai specific.  I think that because of that I never even used log4j - I used the built in (lame) logger instead.  At this point I am guessing that the @Slf4j stuff is pretty universal so I might want to go through and add it across tsugi-util - but I would rather do that as a JIRA on its own because it touches a *lot* of code.

Reply:
you don't have to use slf4j you can use the logging provided by the jdk if that helps!

In any event just asking for log message here vs nothing

Reply:
See the previous comment - going to fix this all in a separate JIRA - https://sakaiproject.atlassian.net/browse/SAK-47912

Reply:
Changed my mind - gonna just add the debug.

Reply:
I had updated the logging a log time ago and forgot I did it :)